image_writer.py

```diff
@@ -61,21 +69,110 @@ def _measure_image(image):
-    draw.multiline_text((x - 3, y + 3), text, font=font, fill=border_color, spacing=spacing)
-    draw.multiline_text((x + 3, y + 3), text, font=font, fill=border_color, spacing=spacing)
+    draw.multiline_text(
+        (x - 3, y - 3), text, font=font, fill=border_color, spacing=spacing)
+    draw.multiline_text(
+        (x + 3, y - 3), text, font=font, fill=border_color, spacing=spacing)
+    draw.multiline_text(
+        (x - 3, y + 3), text, font=font, fill=border_color, spacing=spacing)
+    draw.multiline_text(
+        (x + 3, y + 3), text, font=font, fill=border_color, spacing=spacing)
+
+
+def _fit_text_to_image(image, text_top, text_bottom, draw, font):
+    """Pipes the processes to measure, shrink and wrap text so that it fits within image."""
+    image_w, image_h = _measure_image(image)
+    top_text_fitted, top_font = _shrink_and_wrap_text(draw, text_top, image_w, image_h, font)
+    bottom_text_fitted, bottom_font = _shrink_and_wrap_text(draw, text_bottom, image_w, image_h, font)
+    return top_text_fitted, bottom_text_fitted, top_font, bottom_font
+
+
+def _shrink_and_wrap_text(draw, text, image_w, image_h, normal_font):
+    """Tests the text width and applies alterations to make it fit into the image. First shrinking it, then wrapping,
+    then shrinking and wrapping."""
+    needs_shrinking = _find_if_needs_fitting(draw, text, normal_font, image_w,
+                                             image_h)
+    if needs_shrinking is True:
+        updated_font = ImageFont.truetype('impact.ttf', 33)
+        needs_wrapping = _find_if_needs_fitting(draw, text, updated_font, image_w,
+                                                image_h)
+        if needs_wrapping is True:
+            wrapped_text = _wrap_text(draw, text, normal_font, image_w)
+            needs_shrinking = _find_if_needs_fitting(draw, wrapped_text, normal_font, image_w, image_h)
+            if needs_shrinking is True:
+                wrapped_text = _wrap_text(draw, text, updated_font, image_w)
+                final_font = updated_font
+            else:
+                final_font = normal_font
+        else:
+            wrapped_text = text
+            final_font = updated_font
+    else:
+        wrapped_text = text
+        final_font = normal_font
+    return wrapped_text, final_font
+
+
+def _find_if_needs_fitting(draw, text, font, image_w, image_h):
+    """Determines whether the text is too large, returns bool value True if it is."""
+    text_w, text_h = _measure_text(draw, text, font, SPACING)
+    if text_w > image_w or text_h > image_h // 4:
+        needs_fitting = True
+    else:
+        needs_fitting = False
+    return needs_fitting
+
+
+def _wrap_text(draw, text, font, image_w):
+    """Wraps text around image width, joins it with new line characters."""
+    wrapped_text = intelli_draw(draw, text, font, image_w, SPACING)
+    return '\n'.join(wrapped_text)
+
+
+def intelli_draw(drawer, text, font, image_w, spacing):
```

Comment:
I think why I decided to not put it in a while loop is that I wanted the order of operations to be Shrink -> Wrap (No Shrink) -> Shrink and Wrap rather than it be Shrink -> Shrink and Wrap which seems to be the only option with a while loop. One of the reasons I also don't need to continue shrinking and wrapping is that I know the general range of my max length based on Katie's coding and know I shouldn't need to continue shrinking it.